components/webFeatures.ts

```diff
@@ -217,4 +247,344 @@ webFeaturesRouter.get(
+    }
+
+    let page = 0
+    let hasMore = true
+
+    while (hasMore) {
+        const [results, more] = await getHitsCategory(
+            auth,
+            remoteService,
+            category,
+            page,
+        )
+
+        data.PeacockEscalations = {
+            ...data.PeacockEscalations,
+            ...results.PeacockEscalations,
+        }
+
+        data.PeacockCompletedEscalations = [
+            ...data.PeacockCompletedEscalations,
+            ...results.PeacockCompletedEscalations,
+        ]
+
+        page++
+        hasMore = more
+    }
+
+    return data
+}
+
+webFeaturesRouter.post(
+    "/sync-progress",
+    commonValidationMiddleware,
+    async (req: CommonRequest, res) => {
+        const remoteService = getRemoteService(req.query.gv)
+        const auth = userAuths.get(req.query.user)
+
+        if (!auth) {
+            formErrorMessage(
+                res,
+                "Failed to get official authentication data. Please connect to Peacock first.",
+            )
+            return
+        }
+
+        const userdata = getUserData(req.query.user, req.query.gv)
+
+        try {
+            // Challenge Progression //
+            const challengeProgression = await auth._useService<
+                ChallengeProgressionData[]
+            >(
+                `https://${remoteService}.hitman.io/authentication/api/userchannel/ChallengesService/GetProgression`,
+                false,
+                {
+                    profileid: req.query.user,
+                    challengeids: controller.challengeService.getChallengeIds(
+                        req.query.gv,
+                    ),
+                },
+            )
+
+            userdata.Extensions.ChallengeProgression = Object.fromEntries(
+                challengeProgression.data.map((data) => {
+                    return [
+                        data.ChallengeId,
+                        {
+                            Ticked: data.Completed,
+                            Completed: data.Completed,
+                            CurrentState:
+                                (data.State["CurrentState"] as string) ??
+                                "Start",
+                            State: data.State,
+                        },
+                    ]
+                }),
+            )
+
+            // Profile Progression //
+            const exts = await auth._useService<OfficialProfileResponse>(
+                `https://${remoteService}.hitman.io/authentication/api/userchannel/ProfileService/GetProfile`,
+                false,
+                {
+                    id: req.query.user,
+                    extensions: [
+                        "achievements",
+                        "friends",
+                        "gameclient",
+                        "gamepersistentdata",
+                        "opportunityprogression",
+                        "progression",
+                    ],
```

Comment:
Maybe also get defaultloadout ?

Reply:
Good point, I just ignored that because loadout profiles. One moment please.